File: C/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    // Get Input
    string text = get_string("Text: ");

    // Count letters
    int i = 0;
    int l = 0; // letters
    int w = 1; // words
    int s = 0; // sentences
    do // Until end of string
    {
        // Count letters
        if ((text[i] > 64 && text[i] < 91) || (text[i] > 96 && text[i] < 123))
        {
            l++;
        }
        // Count words
        if (text[i] == 32)
        {
            w++;
        }
        // Count sentences
        if ((text[i] == 33 || text[i] == 46) || text[i] == 63)
        {
            s++;
        }
        i++;
    } while (text[i] > 0);

    // printf("I count %i letters, %i words, and %i sentences\n",l,w,s);

    // Calculate Coleman-Liau index
    float L = l * 100.0 / w;
    float S = s * 100.0 / w;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Output it
    if (index > 15)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n",index);
    }
}

File: Python/sentimental-readability/readability.py
text = input("Text: ")
l = 0
w = 1
s = 0

# Count letters, words, sentences
for i in text:
    letter = ord(i)
    if (letter > 64 and letter < 91) or (letter > 96 and letter < 123):
        l += 1;
    elif letter == 32:
        w += 1
    elif (letter == 33 or letter == 46) or letter == 63:
        s += 1

# Calc CL index
L = l * 100.0 / w
S = s * 100.0 / w
index = round(0.0588 * L - 0.296 * S - 15.8)

if index > 15:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {index}")